File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        temp = {}
        n = len(nums)
        for i in range(n):
            res = target -  nums[i]
            if res in temp:
                return [temp[res],i]
            temp[nums[i]] =  i
        return []
        

        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int , int>hash ;
        for(int i=0;i<nums.size();i++){
            hash[nums[i]] = i;  
        }
        for(int i=0;i<nums.size();i++){
            int com = target - nums[i];
            if(hash.find(com)!= hash.end() && hash[com]!=i){
                return{i, hash[com]};
            }
        }
        return {};
    }
};